### PR TITLE
Add Policy struct to ChainConfig

### PR DESCRIPTION
**Pull request by elmattic**
**Summary of changes**
Changes introduced in this pull request:
- Add network Policy struct into ChainConfig
- Fix existing code to use correct chain Policy

**Reference issue to close (if applicable)**

Closes https://github.com/ChainSafe/forest/issues/1631

**Other information and links**

### CODE REVIEW

File: forest/src/cli/config_cmd.rs [hunk 1]
```diff
@@ -42,6 +42,7 @@ mod tests {
 
         let actual_config: Config = toml::from_str(std::str::from_utf8(sink.buffer()).unwrap())
             .expect("Invalid configuration!");
-        assert_eq!(actual_config, expected_config);
+
+        assert!(expected_config == actual_config);
```
Comment (LesnyRumcajs) [inline forest/src/cli/config_cmd.rs:46]: Hm, why?

File: types/networks/src/lib.rs [hunk 2]
```diff
@@ -134,14 +138,138 @@ struct DrandPoint<'a> {
...
     pub bootstrap_peers: Vec<String>,
     pub block_delay_secs: u64,
     pub version_schedule: Vec<UpgradeInfo>,
     pub height_infos: Vec<HeightInfo>,
+    #[serde(default = "default_policy")]
+    #[serde(with = "serde_policy")]
+    pub policy: Policy,
+}
+
+impl PartialEq for ChainConfig {
```
Comment (LesnyRumcajs) [inline types/networks/src/lib.rs:154]: That's kind of asking for trouble in the future (I'd imagine more fields will come). Probably as a really temporary workaround, it's fine, but should we open a PR adding a proper `PartialEq, Clone` to builtin-actors? I mean, if it's not exposed then it's a really bad API.

File: forest/src/cli/config_cmd.rs (hunk 1 above)
Comment (elmattic) [reply forest/src/cli/config_cmd.rs:46]: This is to avoid to implement `Debug`, needed by `assert_eq`...

Comment (LesnyRumcajs) [reply forest/src/cli/config_cmd.rs:46]: Makes sense. I added the `Debug` in the linked PR as well.

**Review by LesnyRumcajs — APPROVED** [review]
Approving but once `Policy` API is improved in builtin-actors we should revisit those changes. Perhaps a comment somewhere with an issue would be helpful.

File: types/networks/src/lib.rs [hunk 3]
```diff
@@ -134,14 +138,81 @@ struct DrandPoint<'a> {
...
+    fn eq(&self, other: &Self) -> bool {
```
Comment (connormullett) [inline types/networks/src/lib.rs:155]: Can we add a FIXME here for when the builtin-actors PR is good?

**Review by connormullett — CHANGES_REQUESTED** [review]
Looks good! Would like a fixme so we don't lose where that big ole chunky conditional is and what its being blocked by

**Pull request merged**